pandas/_libs/lib.pyx

```diff
@@ -1109,7 +1109,7 @@ _TYPE_MAP = {
     "complex64": "complex",
     "complex128": "complex",
     "c": "complex",
-    "string": "string",
+    str: "string",
```

Comment:
could add this instead of changing.

both StringDtype and ArrowStringDtype have `kind` attrribute as `str`

at the moment the inference matches on `name` so this would need to change for 'string[python]' if we wanted to match on `name`

Reply:
hmm, this doesn't break anything? cause any perf issues?